app/models/group.rb

```diff
@@ -0,0 +1,19 @@
+class Group < ActiveRecord::Base
+
+  belongs_to :creator, class_name: 'User'
+  has_many   :snippets
+
+  validates  :creator, presence: true
+  validates  :name, presence: true
+
+  delegate :username, to: :creator, prefix:true
+
+  def editable_by? user
```

Comment:
This ediable_by / reviewable_by pair of methods would make good candidates to pull out into a module that you would mix into your models.